rfcs/0004-insecure-http/README.md

```diff
@@ -111,9 +111,18 @@ for the required commands, which will be used for specifying the value of `.spec
 > Note: This flag should not be confused with `--insecure-skip-tls-verify` which is meant to skip TLS verification
 > when using an HTTPS connection.
 
+### Proxy
+
+The flag shall also apply to all possible proxy configurations. If the flag `--insecure-allow-http` is set to
+`false`, then specifying the `HTTP_PROXY` environment variable to the controller will lead to the controller
+exiting with a failure on startup.
```

Comment:
Three notes:

1. My concern here would be that `HTTP_PROXY` is generally transparent to the application making HTTP calls (because it's consumed by Go's stdlib). With this change we'll have to inspect `HTTP_PROXY` on startup. It's not dramatic but don't want to leave it unmentioned.
1. What about `HTTPS_PROXY`? You can set `HTTPS_PROXY=http://my-insecure-proxy`. Do we consider this acceptable?
1. What if `HTTP_PROXY=https://my-secure-proxy`? Do we consider this unacceptable, anyway?

Reply:
my thoughts:
1. yeah we'll have to inspect it, i thought that was implied since there is no other way for the controller to be able to fail on startup if its present.
2. no we don't, i'll mention this in the RFC, thanks!
3. `HTTP_PROXY` is not considered when the target URL is `https` so the env var pointing to a `https` url won't have any effect. If the target URL is `http` then the reconciler will fail before it gets to the point where it actually makes the request, so again no effect.